servicecontrol/containerization/index.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Running ServiceControl in containers
+reviewed: 2020-12-10
+---
+
+Docker images for ServiceControl exist on Dockerhub under the [Particular organization](https://hub.docker.com/u/particular). These can be used to run ServiceControl in docker containers. These docker images are only available for Windows.
+
+## Containers overview
+
+ServiceControl is split into multiple docker images for each instance type. These are:
+
+* The error instance
+* The audit instance
+* The monitoring instance
+
+The images for each of these containers are further split into an init container and a runtime container. This means that there are 6 images for all instances of ServiceControl.
+
+The init container is used to create or upgrade the infrastructure required for ServiceControl. This includes creating the database as well as the queues that ServiceControl uses. Once the init container has been run, it will shut down and the runtime container can be run. The runtime container will use the infrastructure that has been created.
+
+### Transports
+
+Each supported transport and topology for ServiceControl has further been broken down into it's own set of images. The following transport and topologies are available on dockerhub:
+
+* [Amazon SQS](https://hub.docker.com/search?q=servicecontrol.amazonsqs&type=image)
+* [Azure Service Bus](https://hub.docker.com/search?q=servicecontrol.azureservicebus&type=image)
+* [Azure Storage Queues](https://hub.docker.com/search?q=servicecontrol.azurestorageque&type=image)
+* [RabbitMQ Conventional Topology](https://hub.docker.com/search?q=servicecontrol.rabbitmq.conven&type=image)
+* [RabbitMQ Direct Topology](https://hub.docker.com/search?q=servicecontrol.rabbitmq.direct&type=image)
+* [Sql Server](https://hub.docker.com/search?q=servicecontrol.sqlserver&type=image)
+
+## Parameters
+
+All parameters that the ServiceControl instance supports can be provided by passing the parameter as an environment variable. E.g.
+
+`-e "ServiceControl/Port=12345"`
```

Comment:
```suggestion
`-e "ServiceControl/Port=12345"`

The available parameters are

- ServiceControl/Port
- ServiceControl/ConnectionString
- ServiceControl.Audit/ConnectionString
- Monitoring/ConnectionString
- ServiceControl/LicenseText

The connectionstring is required by ServiceControl to connect to the transport.

These paramaters can also be added by using a standard Docker environment file. Every parameter has its own line and does not need to be enclosed by  quotes. It can then be used by Docker as follows:

`docker run --env-file servicecontrol.env [dockerimage]`

An example of the servicecontrol.env file:

```
# License text
ServiceControl/LicenseText=<?xml version="1.0" encoding="utf-8"?><license id="..."></license>

# Connection string
ServiceControl/ConnectionString=data source=server,1433; user id=sa; password=[password]; Initial Catalog=servicecontrol
```
```